### ISSUE TITLE
Can't make plantuml-markdown work on Windows

### ISSUE THREAD

**Issue by bcoueraud87** [mikitex70/plantuml-markdown#63]
I am trying to run the command `markdown_py -x plantuml_markdown FMI.md > out.html` on Windows, where FMI.md is a markdown file with a PlantUML diagram inside (attached to this issue). The command works fine on Linux. 

Note that: 

- The PlantUML diagram is valid, as I can extract it from the markdown file and render it with the PlantUML installed on my computer through the help of a `plantuml.bat` script that calls the correct java command.
- The _plantuml-markdown.py scrip_t has also been copied into the _extensions_ directory of the _markdown_ Python package.

[FMI.md](https://github.com/mikitex70/plantuml-markdown/files/8739037/FMI.md)

**Comment by mikitex70** [comment]
Hi @bcoueraud87, what is the behavior?
Is it generating an error, or it simply ignores the diagram?
What version of Python and markdown are you using? I need some info to be able to reproduce your issue.

**Comment by bcoueraud87** [comment]
Hi @mikitex70, on Linux I get the html page with the diagram, whereas on Windows I get an error. Here is the stacktrace I get after running the command `markdown_py -x plantuml_markdown FMI.md > out.html`:

```
Traceback (most recent call last):
  File "c:\program files (x86)\microsoft visual studio\shared\python36_64\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\program files (x86)\microsoft visual studio\shared\python36_64\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\Scripts\markdown_py.exe\__main__.py", line 9, in <module>
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\markdown\__main__.py", line 145, in run
    markdown.markdownFromFile(**options)
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\markdown\core.py", line 407, in markdownFromFile
    kwargs.get('encoding', None))
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\markdown\core.py", line 334, in convertFile
    html = self.convert(text)
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\markdown\core.py", line 261, in convert
    self.lines = prep.run(self.lines)
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\plantuml_markdown.py", line 126, in run
    text1, idx1 = self._replace_block(text[idx:])
  File "C:\Users\coue_be\AppData\Roaming\Python\Python36\site-packages\plantuml_markdown.py", line 222, in _replace_block
    map = etree.fromstring(map_data)
  File "c:\program files (x86)\microsoft visual studio\shared\python36_64\lib\xml\etree\ElementTree.py", line 1315, in XML
    return parser.close()
xml.etree.ElementTree.ParseError: no element found: line 2, column 0
```

My version of Python is 3.6, my version of markdown is 3.3.7 and my version of plantuml-markdown is 3.5.2.

**Comment by mikitex70** [comment]
Hi @bcoueraud87, the issue should now be fixed, install the latest release
There was a bug in the image map detection caused by the different end-of-line terminator between Linux and Windows.
Thanks for reporting it.

**Comment by bcoueraud87** [comment]
Hi @mikitex70, it works, thank you very much!

**Comment by Pfoerd** [comment]
Hi. I just ran into a similar issue on Windows (on Linux everything works as expected). I'm executing exactly what was described in the original issue description above:

`$ markdown_py -x plantuml_markdown -c myconfig.yml FMI.md out.html`

with myconfig.ym being

```
plantuml_markdown:
  format: svg_inline
```

and this always results in the following output

```
Traceback (most recent call last):
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\fesc7420\AppData\Local\Programs\Python\Python38\Scripts\markdown_py.exe\__main__.py", line 7, in <module>
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\markdown\__main__.py", line 145, in run
    markdown.markdownFromFile(**options)
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\markdown\core.py", line 501, in markdownFromFile
    md.convertFile(kwargs.get('input', None),
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\markdown\core.py", line 431, in convertFile
    html = self.convert(text)
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\markdown\core.py", line 354, in convert
    self.lines = prep.run(self.lines)
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\plantuml_markdown\plantuml_markdown.py", line 155, in run
    text1, idx1 = self._replace_block(text[idx:])
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\plantuml_markdown\plantuml_markdown.py", line 224, in _replace_block
    diag_tag = self._image_tag(img_format, diagram, options, code)
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\plantuml_markdown\plantuml_markdown.py", line 234, in _image_tag
    return self._inline_svg_image(diagram, options)
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\site-packages\plantuml_markdown\plantuml_markdown.py", line 252, in _inline_svg_image
    img = etree.fromstring(data.encode('UTF-8'))
  File "c:\users\fesc7420\appdata\local\programs\python\python38\lib\xml\etree\ElementTree.py", line 1320, in XML
    parser.feed(text)
xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 2, column 2
```

As you can see this was executed with python 3.8 but I also tested 3.12. I also tested `encoding: CP1252`. Nothing worked. I also tested different versions of plantuml.

Interesting thing is that when I change the output format to png/svg etc. the build passes but it produces currupt img data so the browser doesn't render them.

- Win10
- $ pip show plantuml-markdown: `Version: 3.9.7`
- Tested with Python 3.8, 3.11 and 3.12


@mikitex70 do you have any idea? I can also open a new issue if this isn't related

**Comment by mikitex70** [comment]
Hi @Pfoerd, I've done some test on Win 10 and I cannot reproduce your issues.
I think the problem may be in the `plantuml` command (it should be a batch file). Locate the `plantuml.bat` and check its contents: the first line MUST be `@echo off`. All the output is captured, so if there is even a single blank line, the output would get corrupted. See the [README.md](https://github.com/mikitex70/plantuml-markdown?tab=readme-ov-file#using-a-local-plantuml-binary) for an example batch file (you can skip the line containing `GRAPHVIZ_DOT`, it is now embedded in the `plantuml.jar`).

The `svg_inline` output is not sensitive to blank lines, but I found a problem with the SVG inserted into the HTML page, which seems to be caused by one of the latest versions of the XML parser. I will fix it soon.

**Comment by mikitex70** [comment]
Released version `3.9.8` which fixes the inline SVG issue.

**Comment by Pfoerd** [comment]
Thank you so much, I really spent hours on this issue. I had another issue with my local configuration and REMd out the `@echo off` for debug purposes. Never though this could cause this issue. And of cause I sent my (corrupted) batch file to all of my colleagues that helped me reproducing the issue 🤣 .

Maybe this could be worth be mentioned in the official docs: 
> the first line MUST be @echo off. All the output is captured, so if there is even a single blank line, the output would get corrupted

Regarding `svg_inline`: `svg_inline` also worked for me after fixing my batch file even without upgrading to `3.9.8`.
